brian2/core/network.py

```diff
@@ -461,6 +464,32 @@ def add(self, *objs):
                         raise TypeError("Can only add objects of type BrianObject, "
                                         "or containers of such objects to Network")
 
+    def get_all_objects(self):
+
+        all_objects = []
+
+        def add_child_objects(*objs):
```

Comment:
I don't think this function should be defined within the `get_all_objects` function (and therefore redefined at each call)

Reply:
Yes, but we want __all_objects__ to be calculated whenever function __get_all_objects()__ is called and we don't store it anywhere, along with we want to have the function to be recursive to traverse heirarchically. 
With these needs, I found difficult to place list (all_objects) intialization and to utilize the list to update with object members with function recursion. I'll make the desirable changes.